src/sinks/influxdb/logs.rs

```diff
@@ -189,24 +250,48 @@ struct InfluxDbLogsEncoder {
     measurement: String,
     tags: HashSet<String>,
     transformer: Transformer,
+    host_key: OwnedValuePath,
+    message_key: OwnedValuePath,
+    source_type_key: OwnedValuePath,
 }
 
 impl HttpEventEncoder<BytesMut> for InfluxDbLogsEncoder {
     fn encode_event(&mut self, event: Event) -> Option<BytesMut> {
         let mut log = event.into_log();
-        log.insert("metric_type", "logs".to_string());
-        let mut log = {
-            let mut event = Event::from(log);
-            self.transformer.transform(&mut event);
-            event.into_log()
-        };
+        // Ensure the "message" isn't overwritten if the event wasn't an object
```

Comment:
This comment is confusing. I'm not sure I understand, or see how the code below is doing this?

I'm guessing we have an event coming in that is just a string:

```
"blahblahblah
```

and we will want the log to then be:

```
{ "message": "blahblahblah" }
```

?

Reply:
I believe it is the case that if we have a log message coming in. This message has a meaning message field of `message`. `self.message_key == zork`. The log message has both fields `message` and `zork` populated. The field `message` will overwrite the field `zork` and the original value of `zork` is lost? 

It might be worth a slight comment to indicate this.

Reply:
If you have an event `. = "zork"` and then `event.insert("foo", "bar")` the result is an event `{ "foo": "bar" }` - I'll expand the comment to better represent the behavior we're avoiding.

Reply:
Updated in e6ee54b21, let me know if that's more clear.